File: 0837-most-common-word/0837-most-common-word.js
/**
 * @param {string} paragraph
 * @param {string[]} banned
 * @return {string}
 */
var mostCommonWord = function(paragraph, banned) {
    const count = {};
    const setBanned = new Set(banned.map(word => word.toLowerCase()));
    const words = paragraph.toLowerCase().split(/[^a-z]+/).filter(word => word);

    let maxCount = 0;
    let res = "";

    for (const word of words) {
        if (!setBanned.has(word)) {
            count[word] = (count[word] || 0) + 1;
            if (count[word] > maxCount) {
                maxCount = count[word];
                res = word;
            }
        }
    }

    return res;
};

File: 0837-most-common-word/0837-most-common-word.py
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        count = defaultdict(int)
        banned = set(word.lower() for word in banned)
        words = re.split(r'[^A-Za-z]+', paragraph.lower())

        maxCout = 0
        res = ""

        for word in words:
            if word and word not in banned:
                count[word] += 1
                if count[word] > maxCout:
                    maxCout = count[word]
                    res = word
        
        return res